crates/bevy_gizmos/src/primitives/dim2.rs

```diff
@@ -857,3 +858,147 @@ where
+pub struct SquircleBuilder<'a, 'w, 's, Config, Clear>
+where
+    Config: GizmoConfigGroup,
+    Clear: 'static + Send + Sync,
+{
+    gizmos: &'a mut Gizmos<'w, 's, Config, Clear>,
+
+    half_size: Vec2, // Half of the size of the squircle
+    p: f32,          // The squareness of the squircle
+
+    position: Vec2, // position of the center of the squircle
+    rotation: f32,  // rotation of the squircle
+    color: Color,   // color of the squircle
+
+    resolution: usize,
+}
+
+impl<'a, 'w, 's, Config, Clear> SquircleBuilder<'a, 'w, 's, Config, Clear>
+where
+    Config: GizmoConfigGroup,
+    Clear: 'static + Send + Sync,
+{
+    /// Set the number of line-segments used to approximate the geometry of this squircle.
+    pub fn resolution(mut self, resolution: usize) -> Self {
+        self.resolution = resolution;
+        self
+    }
+}
+
+impl<'w, 's, Config, Clear> GizmoPrimitive2d<Squircle> for Gizmos<'w, 's, Config, Clear>
+where
+    Config: GizmoConfigGroup,
+    Clear: 'static + Send + Sync,
+{
+    type Output<'a> = SquircleBuilder<'a, 'w, 's, Config, Clear> where Self: 'a;
+
+    fn primitive_2d(
+        &mut self,
+        primitive: &Squircle,
+        position: Vec2,
+        angle: f32,
+        color: impl Into<Color>,
+    ) -> Self::Output<'_> {
+        SquircleBuilder {
+            gizmos: self,
+            half_size: primitive.half_size,
+            p: primitive.p,
+            position,
+            rotation: angle,
+            color: color.into(),
+            resolution: DEFAULT_CIRCLE_RESOLUTION,
+        }
+    }
+}
+
+impl<'a, 'w, 's, Config, Clear> Drop for SquircleBuilder<'a, 'w, 's, Config, Clear>
+where
+    Config: GizmoConfigGroup,
+    Clear: 'static + Send + Sync,
+{
+    fn drop(&mut self) {
+        if !self.gizmos.enabled {
+            return;
+        }
+
+        // We can save on lines by just drawing a rhombus when the squircle is sufficiently similar to a rhombus.
+        if self.p.distance(1.) < 0.0001 {
+            self.gizmos.primitive_2d(
+                &Rhombus {
+                    half_diagonals: self.half_size,
+                },
+                self.position,
+                self.rotation,
+                self.color,
+            );
+        }
+
+        // We can save on lines by just drawing a rectangle when the squircle is sufficiently square.
+        // If `p > 100_000` the difference between a rectangle and a superellipse is less than `0.0000098 * r` for a superellipse with a = b = r.
+        if self.p >= 100_000. {
+            self.gizmos.rect_2d(
+                self.position,
+                self.rotation,
+                self.half_size * 2.,
+                self.color,
+            );
+            return;
+        }
+
+        // Interpolating between `phi`s that are at equal angles from one another and `normal_phi`s
+        // which are spaced so that the normals at each `normal_phi` change by equal amounts gives the most visually consistent look.
```

Comment:
```suggestion
        // which are spaced so that the normals at each `normal_phi` change by equal amounts gives the most visually consistent look.
        // See https://github.com/bevyengine/bevy/pull/13652 for a demonstration
```